raster-foundry/raster-foundry
# Implement custom and single band color mode config UI

## Overview

This PR refactors the custom and single band mode selection to match the formalized API/spec introduced in #2369.

This PR includes a number of hardcoded default color-ramps/maps.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] PR has a name that won't get you publicly shamed for vagueness~

### Demo

![image](https://user-images.githubusercontent.com/2442245/29086161-9dc66b40-7c3f-11e7-9e3e-768d6e44c889.png)


![image](https://user-images.githubusercontent.com/2442245/29086176-ac9123b8-7c3f-11e7-821d-6742cdcdc0d8.png)

### Notes

**On bit depth**
The `colorCorrect.service` has a default bitDepth of 8 defined in https://github.com/raster-foundry/raster-foundry/compare/feature/ak/single-band-raster-ui-2?expand=1#diff-a496a4fe930065dc937fb8ef8285e345R131 

This could be changed. If a color ramp is selected of a certain bit depth and then the bit depth is changed, the front-end will not match it with an existing ramp. I think this is reasonable.

**On custom color composites**
If the user creates a custom color composite that matches an existing preset, on reload, the UI will use the existing preset rather than the custom setting. We have no way of identifying the color composites apart from the band assignments.

## Testing Instructions

 * Start the dev server and the front-end server
 * Open RF and make sure your dev tools are up. We'll be looking at network requests to the `projects` endpoint mostly, so you could filter requests accordingly
 * Go to the color mode section of the project edit UI
 * Test out the custom color mode and ensure it sticks when you change it and reload. (see notes for edge cases)
 * Test out the single band color mode. Make changes and check the network requests to see that they match what you would expect. Especially check that changes made are properly set when you reload the page.

Closes #2328 
Closes #2325
